frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -230,24 +230,34 @@ const GitSection: React.FC<GitSectionProps> = ({
         return;
       }
       const detectedGitType = detectGitType(url);
-      const gitType = values.git.showGitType ? values.git.type : detectedGitType;
       const gitRepoName = formType !== 'sample' && detectGitRepoName(url);
 
       // Updated detectedType only
       if (detectedGitType !== values.git.detectedType) {
-        setFieldValue('git.detectedType', gitType);
+        setFieldValue('git.detectedType', detectedGitType);
       }
       if (detectedGitType === GitTypes.unsure && !values.git.showGitType) {
         setFieldValue('git.showGitType', true);
       }
+
+      if (
+        detectedGitType === GitTypes.github ||
+        detectedGitType === GitTypes.gitlab ||
+        detectedGitType === GitTypes.bitbucket
+      ) {
+        setFieldValue('git.showGitType', false);
+      }
+
+      const gitType = values.git.showGitType ? values.git.type : detectedGitType;
+
       if (gitType !== values.git.type) {
         setFieldValue('git.type', gitType);
       }
 
       const gitService = getGitService(
         url,
         // TODO: ODC-6250 - GitTypes is not compatibily to git service type GitProvider
-        gitType as any,
+        detectedGitType as any,
```

Comment:
I'm not sure, but is this change really needed? Doesn't it make sense to use the selected type if we show this dropdown option?

Reply:
Updated